cdc/sink/ddlsink/mq/kafka_ddl_sink.go

```diff
@@ -127,7 +104,19 @@ func NewKafkaDDLSink(
 		return nil, errors.Trace(err)
 	}
 
-	s, err := newDDLSink(ctx, changefeedID, p, adminClient, topicManager, eventRouter, encoderConfig)
+	syncProducer, err := factory.SyncProducer(ctx)
```

Comment:
why the initializing the SyncProducer should be removed to this location?

Reply:
move it to the last steps, since it's only used to initialize the `ddlPdocuer`